meson.build

```diff
@@ -679,6 +679,10 @@ if time_epoch == -1
         source_date_epoch = run_command('sh', ['-c', 'echo "$SOURCE_DATE_EPOCH"']).stdout().strip()
         if source_date_epoch != ''
                 time_epoch = source_date_epoch.to_int()
+        elif run_command('[', '-d', '.git', ']').returncode() == 0
+                # If we're in a git repository, use the creation time of the latest git tag.
+                latest_tag = run_command('git', 'describe', '--abbrev=0', '--tags').stdout().strip()
+                time_epoch = run_command('git', 'log', '-1', '--format=%at', latest_tag).stdout().to_int()
```

Comment:
i wonder if `git log --tags -1 --format=%at` actually does the same, but in a single line.

I mean, it appears to do that, but I am too dumb to grok the man page if that's actually true... i.e. whether it follows the current branch's history until it finds a tag or does something else... anyone smarter and groks what the man page of git tries to say here?

Reply:
Tags are global I think so it does indeed just find the most recent tag. Could be from any branch though but maybe that's the case with `git describe` as well.

Reply:
Ahah wait it doesn't do exactly the same thing, `git log --tags -1 --format=%at` always returns the timestamp of the latest tag, no matter where in the history we are. So if I checkout v244, the single line command returns the timestamp of v246-rc1 while the describe version returns the timestamp of v244.

I'm not sure if it really matters, but taking where we are in the history into account is probably better than just taking the timestamp of the latest tag.

Reply:
Newer tags should not matter. If I build v244 twice then the same that should be used. It should not matter if another newer tags was created in between the two builds. Anything else breaks reproducibility.